javascript/modules/client/src/client-allowlist.ts

```diff
@@ -0,0 +1,358 @@
+    } else if (!signer.provider) {
+      throw new Error("A web3 provider is needed");
+    }
+
+    // TODO: Implement
+
+    yield {
+      key: ExecuteProposalStep.EXECUTING,
+      txHash: '0x0123456789012345678901234567890123456789012345678901234567890123'
+    }
+    yield {
+      key: ExecuteProposalStep.DONE
+    }
+  }
+
+  private async *_setPluginConfig(_config: VotingConfig): AsyncGenerator<SetVotingConfigStepValue> {
+    const signer = this.web3.getConnectedSigner();
+    if (!signer) {
+      throw new Error("A signer is needed");
+    } else if (!signer.provider) {
+      throw new Error("A web3 provider is needed");
+    }
+
+    // TODO: Implement
+
+    yield {
+      key: SetVotingConfigStep.CREATING_PROPOSAL,
+      txHash: '0x0123456789012345678901234567890123456789012345678901234567890123'
+    }
+    yield {
+      key: SetVotingConfigStep.DONE
+    }
+  }
+
+  private _getMemebers(): Promise<string[]> {
+    const mockAddresses: string[] = [
+      "0x0123456789012345678901234567890123456789",
+      "0x1234567890123456789012345678901234567890",
+      "0x2345678901234567890123456789012345678901",
+      "0x3456789012345678901234567890123456789012",
+      "0x4567890123456789012345678901234567890123",
+    ]
+
+    // TODO: Implement
+
+    return new Promise(resolve => setTimeout(resolve, 1000)).then(() =>
+      mockAddresses.filter(() => Random.getFloat() > 0.4)
+    );
+  }
+
+  private _getProposal(proposalId: string): Promise<AllowListProposal> {
+    if (!proposalId) {
+      throw new Error("Invalid proposalId");
+    }
+
+    // TODO: Implement
+
+    const proposal = getAllowListProposalsWithStatus([getDummyAllowListProposal(proposalId)])[0]
+    return new Promise((resolve) => setTimeout(resolve, 1000)).then(() => (proposal))
+  }
+
+  /**
+   * Returns a list of proposals filtered by the input params
+   *
+   * @private
+   * @param {IProposalQueryParams} {
+   *     limit = 0,
+   *     // TODO
+   *     // uncomment this
+   *     // skip = 0,
+   *     // sortDirection = SortDireccions.ASC,
+   *     // sortBy = AllowListProposalSortBy.CREATED_AT
+   *   }
+   * @return {*}  {Promise<AllowListProposal[]>}
+   * @memberof ClientAllowList
+   */
+  private _getProposalMany({
+    // TODO 
+    // uncomment when querying to subgraph
+    // daoAddressOrEns,
+    limit = 0,
+    // skip = 0,
+    // sortDirection = SortDireccions.ASC,
+    // sortBy = AllowListProposalSortBy.CREATED_AT
+  }: IProposalQueryParams): Promise<AllowListProposal[]> {
+    let proposals: AllowListProposal[] = []
+
+    // TODO: Implement
+
+    for (let index = 0; index < limit; index++) {
+      proposals.push(getDummyAllowListProposal())
+    }
+    proposals = getAllowListProposalsWithStatus(proposals)
+    return new Promise((resolve) => setTimeout(resolve, 1000)).then(() => (proposals))
+  }
+
+  private _buildActionInit(params: IAllowListFactoryParams): PluginInitAction {
+    return {
+      id: this._pluginAddress,
```

Comment:
This one is tricky. One thing is the concrete DAO's Plugin instance, with its own address. A different one is the plugin ID, with is the unique ID that this plugin has on the registry. We should have something like `const PLUGIN_ID = "0x1234...";` at the top of the file and use it here. Same for ERC20